server/api/middleware.go

```diff
@@ -153,3 +153,30 @@ func (s *auditMiddleware) ServeHTTP(w http.ResponseWriter, r *http.Request, next
 		backend.ProcessHTTPRequest(r)
 	}
 }
+
+type rateLimitMiddleware struct {
+	svr *server.Server
+}
+
+func newRateLimitMiddleware(s *server.Server) negroni.Handler {
+	return &rateLimitMiddleware{svr: s}
+}
+
+// ServeHTTP is used to implememt negroni.Handler for rateLimitMiddleware
+func (s *rateLimitMiddleware) ServeHTTP(w http.ResponseWriter, r *http.Request, next http.HandlerFunc) {
+	if !s.svr.GetServiceMiddlewarePersistOptions().IsRateLimitEnabled() {
```

Comment:
Was it possible that the value of `s.svr.GetServiceMiddlewarePersistOptions().IsRateLimitEnabled() ` changed from false to true between `requestInfoMiddleware.ServeHTTP` and here? For example , it is false in the beginning of the `requestInfoMiddleware.ServeHTTP` so that the regionInfo was not created, and it turns true here...

Reply:
If that happens, it will panic on L177 with requestInfo is nil